server/ServerApp.cpp

```diff
@@ -1621,6 +1621,28 @@ bool ServerApp::IsAuthSuccessAndFillRoles(const std::string& player_name, const
     return result;
 }
 
+void ServerApp::AddObserverPlayerIntoGame(const PlayerConnectionPtr& player_connection) {
+    std::map<int, PlayerInfo> player_info_map = GetPlayerInfoMap();
+
+    Networking::ClientType client_type = player_connection->GetClientType();
+    bool use_binary_serialization = player_connection->ClientVersionStringMatchesThisServer();
+
+    if (client_type == Networking::CLIENT_TYPE_HUMAN_OBSERVER ||
```

Comment:
explain the logic here

Reply:
It simply sends `GAME_START` message to established observer player so he knows he's in the game.

Reply:
Add a comment saying that that is what is happening

Reply:
Added comment to the header and in the body of function.